streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBGenericOptionsToDbOptionsColumnFamilyOptionsAdapter.java

```diff
@@ -369,8 +377,9 @@ public Options setMaxBackgroundCompactions(final int maxBackgroundCompactions) {
     }
 
     @Override
-    public void setMaxSubcompactions(final int maxSubcompactions) {
+    public Options setMaxSubcompactions(final int maxSubcompactions) {
```

Comment:
It is orthogonal to this PR, but I've some personal opinions on whether Streams eat this complexity of maintaining the wrapper (and hence each time its upgraded we need to make changes like this , or we should just push this burden to users to distinguish DBOptions and CFOptions): https://issues.apache.org/jira/browse/KAFKA-7869

@mjsax has expressed his preference clearly :) and I'd like to hear other's thoughts: if people favor the first option, then I can go ahead and close that ticket.

@cadonna @ableegoldman @bbejeck @vvcephei @abbccdda

Reply:
Hm. On the one hand this did feel like rather a lot of trivial work (although if we try to keep up with Rocks releases as we go in the future it wouldn't be so bad :) ) and it's not that hard to figure out, especially using an IDE with a good autocomplete.
I'm actually not too concerned about putting the burden on the users, but on the other hand imagine someone has an existing app where they set a large number of options using the ConfigSetter. They'd have to go back and change all of them by hand...personally I'd be a bit annoyed.

Reply:
Actually I just thought of something: if the user constructs a new RocksObject and passes it to options we will never close it and thus leak memory :(

I see two options for us:
1)) use the wrapper to capture all objects created in ConfigSetter and close them ourselves RocksDBStore#close
2) add a #close() method to the ConfigSetter interface and tell users to close any new objects themselves

I think 2) sounds far more reasonable for us to manage but I am worried users won't see this and continue to leak. We'd probably want to cherry-pick this fix back to older branches..?
WDYT @guozhangwang